crates/plugins/src/uninstall.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{get_manifest_file_name, PLUGIN_MANIFESTS_DIRECTORY_NAME};
+use anyhow::Result;
+use std::{fs, path::PathBuf};
+
+/// Settings for uninstalling a plugin.
+pub struct PluginUninstaller {
+    /// Name of plugin to be uninstalled.
+    plugin_name: String,
+    /// Path to the directory where plugins are installed.
+    plugins_dir: PathBuf,
+}
+
+impl PluginUninstaller {
+    pub fn new(plugin_name: &str, plugins_dir: PathBuf) -> Self {
+        Self {
+            plugin_name: plugin_name.to_owned(),
+            plugins_dir,
+        }
+    }
+
+    /// Uninstalls a plugin with a given name, removing it and it's manifest
+    /// from the local plugins directory.
+    pub fn run(&self) -> Result<()> {
+        // Check if plugin is installed
+        let manifest_file = self
+            .plugins_dir
+            .join(PLUGIN_MANIFESTS_DIRECTORY_NAME)
+            .join(get_manifest_file_name(&self.plugin_name));
+        let plugin_exists = manifest_file.exists();
+        match plugin_exists {
+            // Remove the manifest and the plugin installation directory
+            true => {
+                fs::remove_file(manifest_file)?;
+                fs::remove_dir_all(self.plugins_dir.join(&self.plugin_name))?;
```

Comment:
I think I'm clicking what the layout is, but the information seems decentralised.  Is it worth having a `PluginsStore` object that could provide methods like `manifest_file_path(name)` and `plugin_directory(name)` so we're not having to make sure to join to the right parent every time we construct a path?

Reply:
We have a `PluginStore`! 🛍️